data/src/main/java/org/commcare/dalvik/data/network/NetworkUtil.kt

```diff
@@ -0,0 +1,72 @@
+package org.commcare.dalvik.data.network
+
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonObject
+import kotlinx.coroutines.flow.flow
+import org.commcare.dalvik.domain.model.AbdmErrorModel
+import org.commcare.dalvik.domain.model.HqResponseModel
+import retrofit2.Response
+import timber.log.Timber
+
+class NetworkUtil {
+    companion object {
+        const val BASE_URL = "https://ccind.duckdns.org/abdm/api/"
+        const val TRANSLATION_BASE_URL = "https://raw.githubusercontent.com/"
+        fun getTranslationEndpoint(code:String)=
```

Comment:
As ABDM supports multiple language. So we kept the DEFAULT TRANSLATION i.e ENGLISH in code itself.
Now as the application starts it fetches the translation from the server along with the language code provided from the CC application.
So in case of following scenarios the DEFAULT TRANSLATION present in the application is used.

Translation API fails
Any key that is not present in the translation received from server is picked from the DEFAULT TRANSLATION
So we can say this is the backup for translation support

Reply:
@shubham1g5 Motivation to fetch the language from web is to not rebuild the apk for every change/addition in the languages, since app is in development stage and may see frequent language updates. Also, I believe once fetched,  data is cached locally for the session.

Reply:
It seems dangerous to me to rely on being able to pull github raws as part of the production code strategy. It works and is fine for sharing (and probably for testing) but I don't believe github views using the repo hosting as a CDN as falling within their ToS, and could result in them taking action against the repo (or against our accounts).

Reply:
What approach do you suggest for providing files to the app? Requirement here is that url could be formed on the go based on language code and not a random one generated by CDN.

Reply:
If there's not a strong offline component, I think that Firebase would be the default solution for this kind of static serving, but that'll add some ongoing management costs and unpredictability.

Since this app is currently portable and without a global endpoint today, my first impulse would be to say that having language be something that is encoded remotely (rather than in the app) to provide update-free language expansion is a design constraint that I'd push back on the cost/value math for. Is there a specific reason why it's a necessity rather than a nice-to-have?

Reply:
@ankitasonal Distributing localization files and other app assets outside of the binary is definitely one of CommCare's key values, since the application is so heavily localized, and since other forms of application multimedia are often very large requiring a more thoughtful approach than the Play Store distribution mechanism can provide.

I think we have some pretty specific constraints that make that an important option, though, including most of our users needing a huge array of custom translations for their apps. For almost all of our other Play Store apps where the content is more static we use the inbuilt Android translations mechanism (albeit sometimes with other tricks to make it work better with more regional languages). 

Relying on a secondary localization engine (like CommCare does) has significant drawbacks in terms of tooling support and complexity of maintenance and management, and releasing updates to the play store binaries for an app this small doesn't seem particularly imposing. Is there a reason why this app needs to rely on non-APK translations in particular?

Reply:
Language data is bundled now within the app as part of this PR - https://github.com/dimagi/abdm-app/pull/5